packages/block/src/types.ts

```diff
@@ -130,4 +130,5 @@ export interface JsonHeader {
 
 export interface Blockchain {
   getBlock(hash: Buffer): Promise<Block>
+  cliqueActiveSigners(): Address[]
```

Comment:
ah I needed this for the `validateCliqueDifficulty` method which needs to calculate if the signer is in or out of turn. I thought it would be good in the block header next to the pow difficulty check method, but you are right that it adds to the blockchain interface here. I can move it over to blockchain and leave the interface here as before.

Reply:
Just rechecked, so if this is used for typing in `Block` functions like `async validate(blockchain: Blockchain): Promise<void>` then this change is definitely breaking on backwards-compatibility, so we should find another solution here.

Maybe alternatively just leave the interface as-is and manually check in `validateCliqueDifficulty()` for `blockchain.cliqueActiveSigners()` to be present and if not throw? And additionally add some notes on the `blockchain` consuming methods in `Block` that these methods needs a clique supporting `Blockchain` object if run with a PoA setting? 🤔

Not sure if this is the best/optimal solution though.

Reply:
that sounds like a good solution for now, I like keeping the validation in the block header since it's a logical place to validate the difficulty (although it requires the context of blockchain).

i can be convinced to move it but let's see how the solution looks this way. (added in commit: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1074/commits/d58a901b649940b89c0002941f0813c1a2b6cfd2)